vue-frontend/src/stores/index.js

```diff
@@ -0,0 +1,47 @@
+import { defineStore } from "pinia";
+import axios from "axios";
+import config from "@/config.js";
+
+export const useJobListStore = defineStore("jobList", {
+  state: () => {
+    return {
+      items: null,
+      error: null,
+      isLoading: false,
+    };
+  },
+  actions: {
+    async loadJobs() {
+      return new Promise((resolve) => {
+        this.isLoading = true;
+        this.error = null;
+
+        axios
+          .get(config.API_BASE + "/api/careers")
+          .then((response) => {
+            this.items = this.mapJobIcons(response.data);
+            this.isLoading = false;
+            resolve();
+          })
+          .catch((error) => {
+            this.error = error;
+            this.isLoading = false;
+            resolve();
```

Comment:
Shouldn't we have to use `reject()` here?

Reply:
We are not providing a response or error from the function. Caller is supposed to get it from store once it's done waiting. The reason for this change is to avoid try/catch block when calling from components.